megamek/src/megamek/client/ui/baseComponents/AbstractButtonDialog.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2021 - The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MegaMek.
+ *
+ * MegaMek is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MegaMek is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MegaMek. If not, see <http://www.gnu.org/licenses/>.
+ */
+package megamek.client.ui.baseComponents;
+
+import megamek.client.ui.enums.DialogResult;
+import megamek.common.util.EncodeControl;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.ResourceBundle;
+
+/**
+ * This is the Base Dialog for a dialog with buttons in MegaMek. It extends Base Dialog, and adds a
+ * button panel with base Ok and Cancel buttons. It also includes an enum tracker for the result of
+ * the dialog.
+ *
+ * Inheriting classes must call initialize() in their constructor and override createCenterPane()
+ *
+ * The resources associated with this dialog need to contain at least the following keys:
+ * - "Ok.text" -> text for the ok button
+ * - "Ok.toolTipText" -> toolTipText for the ok button
+ * - "Cancel.text" -> text for the cancel button
+ * - "Cancel.toolTipText" -> toolTipText for the cancel button
+ *
+ * This is directly tied to MekHQ's AbstractMHQButtonDialog, and any changes here MUST be verified
+ * there.
+ */
+public abstract class AbstractButtonDialog extends AbstractDialog {
+    //region Variable Declarations
+    private DialogResult result;
+    //endregion Variable Declarations
+
+    //region Constructors
+    protected AbstractButtonDialog(final JFrame frame, final String name, final String title) {

```

Comment:
Let's add some comments to these constructors to indicate what they actually do, semantically speaking.

Reply:
You mean full JavaDoc comments, short bits about the differences between the three, or?